src/library/ValidatorList/Validator/Utils.tsx

```diff
@@ -1,43 +1,56 @@
-// Copyright 2022 @paritytech/polkadot-staking-dashboard authors & contributors
+// Copyright 2023 @paritytech/polkadot-staking-dashboard authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
 import { u8aToString, u8aUnwrapBytes } from '@polkadot/util';
 
-export const getIdentityDisplay = (identity: any, _superIdentity: any) => {
-  // first check identity of validator
+export const getIdentityDisplay = (_identity: any, _superIdentity: any) => {
+  let displayFinal = '';
+  let foundSuper = false;
 
-  // display.Raw
-  let display = identity?.info?.display?.Raw ?? null;
-  // legal.Raw
-  display = display === null ? identity?.info?.legal.Raw ?? null : display;
-
-  // check if identity has been byte encoded
-  const displayAsBytes = u8aToString(u8aUnwrapBytes(display));
-
-  if (displayAsBytes !== '') {
-    return displayAsBytes;
-  }
-  if (display !== null) {
-    return display;
-  }
-
-  // if still null, check super identity
+  // check super identity exists, get display.Raw if it does
   const superIdentity = _superIdentity?.identity ?? null;
+  const superRaw = _superIdentity?.[1]?.Raw ?? null;
+  const superDisplay = superIdentity?.info?.display?.Raw ?? null;
 
-  // display.Raw
-  display = superIdentity?.info?.display?.Raw ?? null;
-  // legal.Raw
-  display = display === null ? superIdentity?.info?.legal.Raw ?? null : display;
+  // check if super raw has been encoded
+  const superRawAsBytes = u8aToString(u8aUnwrapBytes(superRaw));
 
   // check if super identity has been byte encoded
-  const superIdentityAsBytes = u8aToString(u8aUnwrapBytes(display));
+  const superIdentityAsBytes = u8aToString(u8aUnwrapBytes(superDisplay));
 
   if (superIdentityAsBytes !== '') {
-    return superIdentityAsBytes;
+    displayFinal = superIdentityAsBytes;
+    foundSuper = true;
+  } else if (superDisplay !== null) {
+    displayFinal = superDisplay;
+    foundSuper = true;
+  }
+
+  if (!foundSuper) {
+    // cehck sub identity exists, get display.Raw if it does
+    const identity = _identity?.info?.display?.Raw ?? null;
+
+    // check if identity has been byte encoded
+    const subIdentityAsBytes = u8aToString(u8aUnwrapBytes(identity));
+
+    if (subIdentityAsBytes !== '') {
+      displayFinal = subIdentityAsBytes;
+    } else if (identity !== null) {
+      displayFinal = identity;
+    }
   }
-  if (display !== null) {
-    return display;
+  if (displayFinal === '') {
+    return null;
   }
 
-  return display;
+  return (
+    <>
+      {displayFinal}
+      {superRawAsBytes !== '' ? (
+        <span> / {superRawAsBytes}</span>
+      ) : superRaw !== null ? (
+        <span> / {superRaw}</span>
+      ) : null}
+    </>
```

Comment:
It's better to keep the logic away from the layout as you did for `displayFinal`.
Also, please don't use nested ternary operators as it nearly readable and usually considered as a bad practice.

Reply:
This is code logic from the base repository: https://github.com/paritytech/polkadot-staking-dashboard/blob/main/src/library/ValidatorList/Validator/Utils.tsx
I don't believe we need to make any changes to it, as we will have this file after merging the base repository.
I have included it here to cover one feature for now.